cairo-libs/packages/ics23/src/verify.cairo

```diff
@@ -70,8 +71,35 @@ pub fn verify_existence(
 }
 
 pub fn verify_non_existence(
-    spec: @ProofSpec, proof: @NonExistenceProof, root: @RootBytes, key: @KeyBytes,
-) {}
+    spec: @ProofSpec, proof: @NonExistenceProof, root: @RootBytes, key: KeyBytes,
+) {
+    if let Option::Some(left) = proof.left {
+        verify_existence(spec, left, root, left.key, left.value);
+        assert(
+            spec.key_for_comparison(key) > spec.key_for_comparison(left.key.clone()),
+            ICS23Errors::INVALID_LEFT_KEY_ORDER,
+        )
+    } else if let Option::Some(right) = proof.right {
```

Comment:
```suggestion
    }
    if let Option::Some(right) = proof.right {
```

Otherwise, the `right` proof will be ignored if `left` proof is `.is_some()`.